### PR TITLE
SONARJAVA-1167 Handle annotations everywhere

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

File: java-squid/src/main/java/org/sonar/java/ast/parser/TreeFactory.java [hunk 1]
```diff
@@ -736,12 +737,11 @@ public FormalParametersListTreeImpl prependNewFormalParameter(VariableTreeImpl v
     }
   }
 
-  public FormalParametersListTreeImpl newVariableArgumentFormalParameter(Optional<List<AnnotationTreeImpl>> annotations, InternalSyntaxToken ellipsisToken,
-    VariableTreeImpl variable) {
-    variable.addEllipsisDimension(new ArrayTypeTreeImpl(null, annotations.isPresent() ? annotations.get() : ImmutableList.<AnnotationTreeImpl>of(), ellipsisToken));
+  public FormalParametersListTreeImpl newVariableArgumentFormalParameter(Optional<List<AnnotationTreeImpl>> annotations, InternalSyntaxToken ellipsisToken, VariableTreeImpl variable) {
```
Comment (SonarTech) [inline java-squid/src/main/java/org/sonar/java/ast/parser/TreeFactory.java]: MINOR Split this 184 characters long line (which is greater than 180 authorized). [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00103)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Comment by benzonico** [comment]
LGTM

File: java-checks/src/main/java/org/sonar/java/checks/UselessImportCheck.java [hunk 2]
```diff
@@ -110,6 +110,10 @@ public void scanFile(JavaFileScannerContext context) {
     leaveFile();
   }
 
+  private ExpressionTree getPackageName(CompilationUnitTree cut) {
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/UselessImportCheck.java]: MAJOR Make "getPackageName" a "static" method. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Comment by benzonico** [comment]
LGTM

**Pull request merged**